### PR TITLE
Remove temporary files after use

### PR DESCRIPTION
**Pull request by akodanka**
Deleting the serialized IR files generated by NNHAL, at the end of its usage.

Tracked-On: OAM-106941

### CODE REVIEW

File: IENetwork.cpp [hunk 1]
```diff
@@ -73,10 +72,9 @@ ov::Tensor IENetwork::getOutputTensor(const std::size_t index) {
 }
 
 void IENetwork::infer() {
-    ALOGI("infer Network\n");
-    mInferRequest.start_async();
-    mInferRequest.wait_for(std::chrono::milliseconds(10000));
-    ALOGI("infer request completed");
+    ALOGI("infer requested");
+    mInferRequest.infer();
```
Comment (rairatne) [inline IENetwork.cpp:76]: can we keep it as it is, and just modify wait?

mInferRequest.start_async();
// Wait for inference completion
As mentioned in docs?
https://docs.openvino.ai/latest/openvino_2_0_inference_pipeline.html#doxid-openvino-2-0-inference-pipeline
mInferRequest.wait();

Comment (rairatne) [reply IENetwork.cpp:76]: Have we tried this already and checked whether that segfault is going away with the above modified wait condition

**Review by rairatne — APPROVED** [review]
LGTM apart from the concerns mentioned in comments

**Review by rairatne — APPROVED** [review]
LGTM

**Review by JeevakaPrabu — APPROVED** [review]

**Pull request merged**